tools/clang/unittests/HLSLExec/ShaderOpTest.h

```diff
@@ -218,6 +218,18 @@ class ShaderOp {
     }
     return result;
   }
+  LPCSTR GetShaderArguments(ShaderOpShader *pShader) {
+    if (!pShader || !pShader->Arguments) return nullptr;
+    LPCSTR result = pShader->Arguments;
+    if (result[0] == '@') {
+      for (auto && S : Shaders) {
+        if (S.Name && 0 == strcmp(S.Name, result + 1))
```

Comment:
if `result` is really the list of arguments, are they separated by null terminators? otherwise, I imagine that the names may match, but the result string will be longer than S.name because of the rest of the arguments, and the strcmp function won't return 0.

Reply:
`result` is either the list of arguments separated by spaces or else it is the special code `@<identifier>`.  If it has this special code followed by additional arguments, it's malformed and shouldn't match since we're not going to pay attention to those arguments. To allow that might be useful, but it differs from how the original implementation did things and I personally think it adds as much confusion as convenience.